lightning_client.go

```diff
@@ -3619,6 +3625,20 @@ func (s *lightningClient) ChannelAcceptor(ctx context.Context,
 	return errChan, nil
 }
 
+// FundingStateStep is a funding related call that allows the execution
+// of some preparatory steps for a funding workflow or manual
+// progression of a funding workflow.
+func (s *lightningClient) FundingStateStep(ctx context.Context,
```

Comment:
I'm wondering if it's better to keep this call as a wrapper around the raw rpc client call or create some higher level abstraction that fits golang better? Maybe @guggero can also chime in here since funding is his expertise.

Reply:
I think it's okay to have the light wrapper as it's currently implemented in this PR. Otherwise we'd need an enum or multiple structs to abstract the different messages that can be sent as a funding state step. And I don't really see the benefit in that.
We probably should just add all existing RPCs to `lndclient` so we can deprecate the use of the basic client.